### PR TITLE
Update Using Chapel "prerequisites" list

### PR DESCRIPTION
**Pull request by awallace-cray**
* `Mason` almost always requires git; `mason system` requires pkg-config.
* Recommended `FreeBSD` packages are updated consistent with current Vagrant tests. 
* The `Fedora 21/Fedora 22` distinction is obsolete; that was 2015.

### CODE REVIEW

**Comment by ronawho** [comment]
Update the ` * If you wish to use the ``mason system``, chapel's package manager, ``pkg-config`` is required.` note to include `git`?

**Comment by ronawho** [comment]
I don't feel particularly qualified to review this. @ben-albrecht and/or @Spartee are probably better candidates.

**Comment by awallace-cray** [comment]
I added 'git' to the mason requirements. All I know about it is one of the unit tests broke when I was running the pre-release tests on a Docker image which did not have "git".  Seems like it does no harm.

File: doc/rst/usingchapel/prereqs.rst [hunk 1]
```diff
@@ -40,8 +40,8 @@ about your environment for using Chapel:
   * If you wish to use chpldoc or Chapel's test system, ``curl`` and
     python-devel (or equivalent packages for your platform) are required.
 
-  * If you wish to use the ``mason system``, chapel's package manager, ``pkg-config`` is
-    required.
+  * If you wish to use the ``mason system``, chapel's package manager, ``pkg-config`` and
+    ``git`` are required.
```
[the remark this answers is not in the document]
Comment (ben-albrecht) [reply doc/rst/usingchapel/prereqs.rst]: Is `git` not required for typical `mason` usage? For example, `mason new` creates a new git repo, and packages are pulled via `git`.

[the remark this answers is not in the document]
Comment (Spartee) [reply doc/rst/usingchapel/prereqs.rst]: Because we added the options for offline usage such as `--no-init` for `mason new` and `--no-update-registry` for `mason update` and `mason build`, technically `git` is not required for building libs/apps without dependencies. 

I might mention a line that says something about the functionality of mason being significantly limited without `git`. 

We might consider making an `MASON_OFFLINE` mode where those arguments are added automatically.

[the remark this answers is not in the document]
Comment (ben-albrecht) [reply doc/rst/usingchapel/prereqs.rst]: I see. I suspect there may be some cases where we would be using `git` even in offline mode, e.g. checking out a different version of a pre-installed library.

For now, it may be safer to state Mason depends on `git` until we have an offline "mode" available via environment variable, and have sufficiently tested `git`-less mason usage.

[the remark this answers is not in the document]
Comment (ben-albrecht) [reply doc/rst/usingchapel/prereqs.rst]: `git` dependency aside, I'd suggest rewording this sentence:

```
* If you wish to use :ref:`readme-mason`, chapel's package manager,  ``git`` is required.
    - The ``mason system`` subcommands additionally require ``pkg-config`` 
```

**Comment by awallace-cray** [comment]
@ben-albrecht Thanks for your improvement, which I just implemented.

**Comment by awallace-cray** [comment]
Oops, think I'd better do the make docs thing on this.
